### DISCUSSION TITLE
How to wait for a new index to be ready

### DISCUSSION THREAD

**Discussion by tzarger** [ravendb/ravendb#15806] [Q&A]
We are creating an index then later adding some more fields, and when patching the data right after programmatically updating the index, the side-by-side index is created and then eventually when the side-by-side index is done, then replaces the original index.

Since we need to immediately run a patch, the field we are patching does not exists yet since the side-by-side index is not done.

How can we know then the side-by-side index is done and the new fields added are there? Or how can we wait for that?

Hope this makes sense.

Thanks!

**Comment by ayende** [comment]
You can get the indexing status. The easiest way is to check the if a side by side index exists and poll it every few minutes.
There is the `GetIndexesStatisticsOperation` you can use for that.
Check if there is a result for side by side and if the main one `IsStale` is.
